### PR TITLE
fixed issue 37 and 10

### PR DESCRIPTION
**Pull request by Flippchen**

### CODE REVIEW

**Review by Nereuxofficial — APPROVED** [review]
Coole Error Messages!

File: lib/comms.dart [hunk 1]
```diff
@@ -13,7 +13,8 @@ Future<bool> joinEvent(int id) async {
       await http.put(Uri.parse("$server_url/join_event/$id"), headers: {
     "cookie": "user_token=${userToken!}",
   });
-
+  print("body:");
```
Comment (Flippchen) [inline lib/comms.dart:16]:
```suggestion
```

File: lib/comms.dart [hunk 2]
```diff
@@ -13,7 +13,8 @@ Future<bool> joinEvent(int id) async {
...
+  print(response.body);
```
Comment (Flippchen) [inline lib/comms.dart:17]:
```suggestion
```

**Review by MarionHinkel — APPROVED** [review]
LGTM

File: lib/my_event_element.dart [hunk 3]
```diff
@@ -66,29 +66,30 @@ class _MyEventState extends State<MyEvent> {
...
-                      });
-                  var jsnondecoded = json.decode(event.body);
-                  var creator = jsnondecoded["creator"];
-                  UserData user = await getUserId();
-                  var uid = user.id;
-                  if (creator == uid) {
-                    showAlertDialog(context, "You could not leave the event.",
-                        "You are the owner of the event and therefore cannot leave the event.\nYou can delete the event instead.");
-                  } else {
+                //-----------------------------------
```
Comment (Flippchen) [inline lib/my_event_element.dart:69]:
```suggestion
```

File: lib/my_event_element.dart [hunk 4]
```diff
@@ -66,29 +66,29 @@ class _MyEventState extends State<MyEvent> {
...
+                } else {
+                  var hasLeft = await leaveEvent(widget.data.id);
+                  if (hasLeft)
+                    widget.callback();
+                  else
                     showAlertDialog(context, "Unable to leave the event.",
                         "There was an unexpected error.\n\nPlease try again later.");
-                  }
                 }
+                //-----------------------------------
```
Comment (Flippchen) [inline lib/my_event_element.dart:91]:
```suggestion
```

**Review by mfloto — APPROVED** [review]
LGTM

**Pull request merged**